src/gui/qgsattributeform.cpp

```diff
@@ -2429,6 +2440,14 @@ void QgsAttributeForm::afterWidgetInit()
       connect( eww, &QgsEditorWidgetWrapper::valuesChanged, this, &QgsAttributeForm::onAttributeChanged );
       connect( eww, &QgsEditorWidgetWrapper::constraintStatusChanged, this, &QgsAttributeForm::onConstraintStatusChanged );
     }
+    else
+    {
+      QgsRelationWidgetWrapper *relationWidgetWrapper = qobject_cast<QgsRelationWidgetWrapper *>( ww );
+      if ( relationWidgetWrapper )
+      {
+        connect( relationWidgetWrapper, &QgsRelationWidgetWrapper::relatedFeaturesChanged, this, &QgsAttributeForm::onRelatedFeaturesChanged, Qt::QueuedConnection );
```

Comment:
Would it make sense to combine  `Qt::QueuedConnection | Qt::UniqueConnection`?

Reply:
This connect is called only once but i can add it in case something change later... some lines above there are more `connect` I will add it there too